### PR TITLE
 Move language analyzers from server to analysis-common module.

### PR DESCRIPTION
**Pull request by martijnvg**
The following analyzers were moved from server module to analysis-common module:
`greek`, `hindi`, `hungarian`, `indonesian`, `irish`, `italian`, `latvian`,
`lithuanian`, `norwegian`, `persian`, `portuguese`, `romanian`, `russian`,
`sorani`, `spanish`, `swedish`, `turkish` and `thai`.

Relates to #23658

### CODE REVIEW

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java [hunk 1]
```diff
@@ -262,7 +302,7 @@
         analyzers.add(new PreBuiltAnalyzerProviderFactory("bulgarian", CachingStrategy.LUCENE, BulgarianAnalyzer::new));
         analyzers.add(new PreBuiltAnalyzerProviderFactory("catalan", CachingStrategy.LUCENE, CatalanAnalyzer::new));
         // chinese analyzer: only for old indices, best effort
-        analyzers.add(new PreBuiltAnalyzerProviderFactory("chinese", CachingStrategy.LUCENE, StandardAnalyzer::new));
+        analyzers.add(new PreBuiltAnalyzerProviderFactory("chinese", CachingStrategy.ONE, StandardAnalyzer::new));
```
Comment (nik9000) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: Did you mean to change this one?

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java [hunk 2]
```diff
@@ -247,13 +285,15 @@
     @Override
     public List<PreBuiltAnalyzerProviderFactory> getPreBuiltAnalyzerProviderFactories() {
         List<PreBuiltAnalyzerProviderFactory> analyzers = new ArrayList<>();
-        analyzers.add(new PreBuiltAnalyzerProviderFactory("standard_html_strip", CachingStrategy.LUCENE,
+        analyzers.add(new PreBuiltAnalyzerProviderFactory("standard_html_strip", CachingStrategy.ELASTICSEARCH,
```
Comment (nik9000) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: Did you mean to change this one?

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java (hunk 1 above)
Comment (martijnvg) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: Yes, I made a copy paste error. If you take a look at the history of the `PreBuiltAnalyzers.java` file, you will see that this was the cache strategy used. Sorry, I should have mentioned this specifically in this pr.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java (hunk 2 above)
Comment (martijnvg) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: yes, see previous comment.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**